python/ray/dag/tests/experimental/test_accelerated_dag.py

```diff
@@ -1134,6 +1137,54 @@ def test_intra_process_channel_with_multi_readers(self, ray_start_regular):
         compiled_dag.teardown()
 
 
+class TestLeafNode:
+    def test_leaf_node_one_actor(self, ray_start_regular):
```

Comment:
Can you add a test which checks if this works? The code is taken from existing dag test (for checking if scanner works properly). If you plan to address apply_recursive right after this PR, we don't need this

```
def test_output_node(shared_ray_instance):
    @ray.remote
    def f(input):
        return input

    with pytest.raises(ValueError):
        with InputNode() as input_data:
            dag = MultiOutputNode(f.bind(input_data))

    with InputNode() as input_data:
        dag = MultiOutputNode([f.bind(input_data)])

    assert ray.get(dag.execute(1)) == [1]
    assert ray.get(dag.execute(2)) == [2]

    with InputNode() as input_data:
        dag = MultiOutputNode([f.bind(input_data["x"]), f.bind(input_data["y"])])

    refs = dag.execute({"x": 1, "y": 2})
    assert len(refs) == 2
    assert ray.get(refs) == [1, 2]

    with InputNode() as input_data:
        dag = MultiOutputNode(
            [f.bind(input_data["x"]), f.bind(input_data["y"]), f.bind(input_data["x"])]
        )

    refs = dag.execute({"x": 1, "y": 2})
    assert len(refs) == 3
    assert ray.get(refs) == [1, 2, 1]
```

Reply:
what's the relationship between this test and scanner?

Reply:
> If you plan to address apply_recursive right after this PR, we don't need this

I will not update `apply_recursive` in this PR because it is on Ray Serve's codepath.

Reply:
just want to make sure indexing inputs works properly. I think this is handled by apply_recursive before, and we need to make sure the new bfs logic can handle the same thing. 

```
with Inputnode() as inp:
    a = inp["a"]
    b = inp["b"]
    ...

adag.execute({"a": 1, "b": 2})
```

Reply:
I still don't fully understand the relationship between this test and scanner. Maybe we can chat tmr.

Added tests: https://github.com/ray-project/ray/pull/47261/commits/eb1914c73f56cef6899b75910a530b916ff7d0b0

The test in this PR is a bit different from the test in `test_output_node.py`.

1. ADAG doesn't support function node.
2. A shared memory channel doesn't support to be read by the same actor more than once, so I use multiple actors in this case.
3. The way to index input is different from DAG and ADAG. For example,

https://github.com/ray-project/ray/blob/650bd2672d52341c549e5a640b1f6dfa06a1869f/python/ray/dag/tests/experimental/test_accelerated_dag.py#L357-L364